packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -207,6 +211,8 @@ class Dart2JSTarget extends Target {
         '-Ddart.vm.product=true',
       for (final String dartDefine in decodeDartDefines(environment.defines, kDartDefines))
         '-D$dartDefine',
+      if (!sourceMapsEnabled)
+        '--no-source-maps',
```

Comment:
Should we pass `--source-maps` when `sourceMapsEnabled` for explicitness?

Reply:
--source-maps is not a flag, just --no-source-maps